src/silo/database.cpp

```diff
@@ -422,68 +474,80 @@ void Database::saveDatabaseState(const std::string& save_directory) {
    });
    SPDLOG_INFO("Finished saving partitions", partitions.size());
 
-   saveDataVersion(*this, save_directory);
+   saveDataVersion(*this, versioned_save_directory);
 }
 
-DataVersion loadDataVersion(const std::string& save_directory) {
-   std::ifstream data_version_file(save_directory + "data_version.silo");
-   std::string data_version;
-   data_version_file >> data_version;
-   return DataVersion(data_version);
+DataVersion loadDataVersion(const std::filesystem::path& filename) {
+   std::ifstream data_version_file = openInputFileOrThrow(filename);
+   std::string data_version_string;
+   data_version_file >> data_version_string;
+   auto data_version = DataVersion::fromString(data_version_string);
+   if (data_version == std::nullopt) {
+      auto error = fmt::format(
+         "Data version file {} did not contain a valid data version: {}",
+         filename.string(),
+         data_version_string
+      );
+      SPDLOG_ERROR(error);
+      throw persistence::LoadDatabaseException(error);
```

Comment:
Why do we log and throw the same message? I think we could leave the log part to the place where we catch the exception.